src/main/scala/com/amazon/deequ/repository/MetricsRepositoryDiff.scala

```diff
@@ -0,0 +1,98 @@
+  * permissions and limitations under the License.
+  *
+  */
+
+package com.amazon.deequ.repository
+
+import java.io.{BufferedWriter, File, FileWriter}
+
+import com.amazon.deequ.analyzers.Analyzer
+import com.amazon.deequ.analyzers.runners.{AnalysisRunner, AnalysisRunnerRepositoryOptions, AnalyzerContext}
+import com.amazon.deequ.metrics.Metric
+import com.amazon.deequ.repository.fs.FileSystemMetricsRepository
+import org.apache.spark.sql.{DataFrame, SparkSession}
+import org.apache.spark.storage.StorageLevel
+
+class MetricsRepositoryDiff (session: SparkSession,
+                             path: String,
+                             saveAllRepositories: Boolean = false) {
+
+  private def jsonUnion(jsonOne: String, jsonTwo: String): String = {
+
+    val objectOne: Seq[AnalysisResult] = AnalysisResultSerde.deserialize(jsonOne)
+    val objectTwo: Seq[AnalysisResult] = AnalysisResultSerde.deserialize(jsonTwo)
+
+    val unioned = objectTwo ++ objectOne
+
+    AnalysisResultSerde.serialize(unioned)
+  }
+
+  def saveRepositoryAsJSON(data: DataFrame, analyzers: Seq[Analyzer[_, Metric[_]]]): String = {
+    val timestamp = System.currentTimeMillis()
+    val metricsFile = new File(path, timestamp.toString + "metrics.json")
+
+    val repository: MetricsRepository =
+      FileSystemMetricsRepository(session, metricsFile.getAbsolutePath)
+
+    val resultKey = ResultKey(timestamp, Map("tag" -> "data"))
+
+    val analysisResults = AnalysisRunner.doAnalysisRun(
+      data,
+      analyzers,
+      None,
+      None,
+      StorageLevel.MEMORY_AND_DISK,
+      AnalysisRunnerRepositoryOptions(Some(repository)))
+
+    val currentValueForKey = repository.loadByKey(resultKey)
+
+    // AnalyzerContext entries on the right side of ++ will overwrite the ones on the left
+    // if there are two different metric results for the same analyzer
+    val valueToSave = currentValueForKey.getOrElse(AnalyzerContext.empty) ++ analysisResults
+
+    if(saveAllRepositories) {
+      repository.save(Some(resultKey).get, valueToSave)
+    }
+
+    val successfulMetrics = valueToSave.metricMap
+      .filter { case (_, metric) => metric.value.isSuccess }
+
+    val analyzerContextWithSuccessfulValues = AnalyzerContext(successfulMetrics)
+
+    AnalysisResultSerde.serialize(
+      Seq(AnalysisResult(Some(resultKey).get, analyzerContextWithSuccessfulValues))
+    )
+  }
+
+  def aggregateMultipleMetricsRepositories(datasets: Seq[DataFrame],
+                                           analyzers: Seq[Analyzer[_, Metric[_]]],
+                                           name: String = "multimetricsrepo.json"): Any = {
+    var json: String = ""
+    val multiRepoFile = new File(path, name)
+
+    datasets.foreach { data =>
+      val tempJson = saveRepositoryAsJSON(data, analyzers)
+      if (json == "") {
+        json = tempJson
+      }
+      else {
+        json = jsonUnion(json, tempJson)
+      }
+    }
```

Comment:
Can you implement the union in a way so this conditional is not necessary? So the union works even if one of the inputs is the empty string?

Reply:
The union uses the  AnalysisResultSerde.deserialize function within deequ. This doesn't support the emtpy string as input and as we didn't want to reimplement the whole function just to add that functionality, we decided to skip it on the very first step. This way we also don't have to run the whole function as we can just the initial json variable to tempJson. If you have any suggestions how to better implement it, feel free to leave some feedback!